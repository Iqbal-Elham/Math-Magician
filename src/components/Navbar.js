import React from 'react';
import { NavLink, Outlet } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.css';
import styles from './styles/Navbar.module.css';

const Navbar = () => {
  const clickedStyle = ({ isActive }) => ({
    color: isActive ? 'pink' : 'black',
  });
  return (
    <>
      <nav className={styles.nav}>
        <div>
          <h1>Math Magician</h1>
        </div>
        <div className={styles.navlink}>
          <NavLink style={clickedStyle} to="/">
            Home
          </NavLink>
          <NavLink style={clickedStyle} to="/calculator">
            Calculator
          </NavLink>
          <NavLink style={clickedStyle} to="/quote">
            Quote
          </NavLink>
        </div>
      </nav>
      <Outlet />
    </>
  );
};

export default Navbar;
