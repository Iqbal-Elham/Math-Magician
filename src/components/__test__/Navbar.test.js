import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Navbar from "../Navbar";
​
describe("Navbar", () => {
  it("should return the number navlinks", () => {
    render(
      <Router>
        <Navbar />
      </Router>
    );
    const navlinks = screen.getAllByRole("link");
    expect(navlinks).toHaveLength(3);
  });
​
  it("Should render the user interaction of navbar", () => {
    const tree = renderer.create(
        <Router>
          <Navbar />
        </Router>
    ).toJSON;
    expect(tree).toMatchSnapshot();
  })
});