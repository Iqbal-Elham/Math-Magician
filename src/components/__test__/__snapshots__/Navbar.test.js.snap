// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render the user interaction of navbar 1`] = `[Function]`;
